### DISCUSSION TITLE
Plugin development: Get list of all assets + their original file names

### DISCUSSION THREAD

**Discussion by PixelsCommander** [webpack/webpack#13558] [Q&A] [answered]
Hi guys

I stuck into a problem that might look trivial but after digging a bit I feel like I am going to spend many days on it unless I get a bit of direction from you.

The goal is to have a plugin that saves a key-value list of {assets' old file names: assets' new file names}.

I found out that modules are where I look for original file names and assets are the final list of filenames. Any ideas on how would I link assets back to modules or another way around?

**Comment by alexander-akait** [comment, marked as answer]
You can use this https://github.com/shellscape/webpack-manifest-plugin, most of assets has `asset.info.sourceFilename` https://github.com/webpack-contrib/copy-webpack-plugin/blob/master/src/index.js#L792

**Reply by PixelsCommander** [reply]
Many thanks!
